File: fontes/pascal/micro/micro10.pas
program micro10;

function fatorial (n:integer) : integer;
begin
 	if(n <= 0) then
 		fatorial := 1
 	else
 		fatorial := n * fatorial(n-1);
end;

var
 	numero, fat : integer;
begin
 	write('Digite um numero: ');
 	readln(numero);
 	fat := fatorial(numero);
 	writeln('O fatorial de ', numero, 'eh', fat);
end.

File: fontes/c/micro/micro10.c
#include <stdio.h>
#include <stdlib.h>

int fator(int numero)
{
 if(numero > 1)
 	return numero * fator(numero - 1);
 else
 	return 1;
}

int main()
{
 int numero= 0;
 int fat;
 printf("Digite um numero: ");
 scanf("%d", &numero);
 fat = fator(numero);
 printf("O fatorial de %d é %d\n",numero, fat);
 return 0;
}